System/Random.hs

```diff
@@ -117,6 +183,21 @@ import qualified System.Random.SplitMix as SM
 
 import GHC.Exts (Ptr(..), build, byteArrayContents#, unsafeCoerce#)
 
+-- $setup
+-- >>> import Foreign.Marshal.Alloc (alloca)
+-- >>> import Foreign.Storable (peekByteOff, pokeByteOff)
+-- >>> import System.IO.Unsafe (unsafePerformIO)
+-- >>> :{
+-- unBuildWord32 :: Word32 -> (Word16, Word16)
+-- unBuildWord32 w = unsafePerformIO $ alloca f
```

Comment:
It is better to avoid using unsafe operations in documentation, especially if it can be achieved in a more functional way:
```haskell
unBuildWord32 :: Word32 -> (Word16, Word16)
unBuildWord32 w = (fromIntegral (shiftR 16 w), fromIntegral (fromIntegral (maxBound :: Word16) .&. w))
```